src/Avalonia.Controls/TopLevel.cs

```diff
@@ -194,9 +202,18 @@ public TopLevel(ITopLevelImpl impl, IAvaloniaDependencyResolver dependencyResolv
         public Size ClientSize
         {
             get { return _clientSize; }
-            protected set { SetAndRaise(ClientSizeProperty, ref _clientSize, value); }
+            protected set
+            {
+                SetAndRaise(ClientSizeProperty, ref _clientSize, value);
+                SetAndRaise(TotalSizeProperty, ref _totalSize, PlatformImpl.TotalSize);
```

Comment:
Putting this in the `ClientSize` setter feels a bit hacky, especially as the value passed to this property has nothing to do with the value which will be written to `TotalSize`.